tech_reports/LLMs/llms.md

```diff
@@ -55,14 +55,281 @@ Authors:
 ### 2.7 LM Head
 ## 3. Features
 ### 3.1 Generative Decoding
+
 ### 3.2 Prefill and Decode
-  - submodules, tests
-  - how to combine prefill and decode, 
-  - slicing prefill to fit in L1
+
+Large language models require two distinct phases for inference due to the fundamental nature of transformer attention and autoregressive generation: prefill and decode.
+
+On tenstorrent harware, the prefill phase is done sequentially for each user, but parallel for the prompt tokens of each user. During prefill, the model computes attention scores for all prompt tokens against each other and populates the key-value (KV) cache which will speed up the computation of the decode phase. At the end of the prefill phase, the first token for the following autoregressive generation will also be computed.
```

Comment:
It is not because of the tenstorrent hardware, people might think there is some hardware limitation.

How about "On our LLM implementations, ..."